pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -136,7 +143,30 @@ func (v *fleetResourceValidator) handleRoleBinding(req admission.Request) admiss
 	if err := v.decodeRequestObject(req, &rb); err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
-	return validation.ValidateUserForFleetResource(rb.Kind, types.NamespacedName{Name: rb.Name, Namespace: rb.Namespace}, v.whiteListedUsers, req.UserInfo)
+	return validation.ValidateUserForResource(rb.Kind, types.NamespacedName{Name: rb.Name, Namespace: rb.Namespace}, v.whiteListedUsers, req.UserInfo)
+}
+
+func (v *fleetResourceValidator) handleNamespace(req admission.Request) admission.Response {
+	var currentNS corev1.Namespace
+	if err := v.decodeRequestObject(req, &currentNS); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	fleetMatchResult := regexp.MustCompile(fleetMatch).FindStringSubmatch(currentNS.Name)
+	kubeMatchResult := regexp.MustCompile(kubeMatch).FindStringSubmatch(currentNS.Name)
+	if len(fleetMatchResult) > 0 {
+		if req.Operation == admissionv1.Update {
+			var oldNS corev1.Namespace
+			if err := v.decoder.DecodeRaw(req.OldObject, &oldNS); err != nil {
+				return admission.Errored(http.StatusBadRequest, err)
+			}
+			return validation.ValidateMCOrNSUpdate(&currentNS, &oldNS, v.whiteListedUsers, req.UserInfo)
```

Comment:
Hi Arvind! Not sure if I misunderstand anything here, but for namespaces, it's more of an all-or-nothing scenario I suppose. If the user is whitelisted, they can do anything to the object; otherwise nothing is allowed -> i.e., no need to check if the labels/annotations are changed, etc.